### PR TITLE
Shooter

### PR DESCRIPTION
**Pull request by UnknownSilicon**
Ignore Travis, once the code is merged it should work. It's a commons library problem

### CODE REVIEW

File: src/org/robockets/steamworks/PortNumbers.java [hunk 1]
```diff
@@ -10,19 +10,20 @@
...
     public static final int DRIVETRAIN_FRONT_LEFT_SC_PORT = 0;
     public static final int DRIVETRAIN_FRONT_RIGHT_SC_PORT = 1;
     public static final int DRIVETRAIN_BACK_LEFT_SC_PORT = 2;
     public static final int DRIVETRAIN_BACK_RIGHT_SC_PORT = 3;
-
-    public static final int SHOOTER_CONVEYOR_SC_ONE = 5;
-    public static final int SHOOTER_CONVEYOR_SC_TWO = 6;
     
-    public static final int BALL_INTAKE_ROLLER_SC_PORT = 7;
+    public static final int BALL_INTAKE_ROLLER_SC_PORT = 8;
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/PortNumbers.java]: I believe I set the climber port to 8 in the climber branch.

File: src/org/robockets/steamworks/commands/KillShooter.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package org.robockets.steamworks.commands;
+
+import edu.wpi.first.wpilibj.command.Command;
+
+import org.robockets.steamworks.Robot;
+
+/**
+ * @author Jake Backer
+ */
+public class KillShooter extends Command {
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/commands/KillShooter.java:10]: Let's try to test not using this command first. If WPILib's `end()` method is sufficient, then we should go with than.

File: src/org/robockets/steamworks/subsystems/Shooter.java [hunk 3]
```diff
@@ -16,10 +20,11 @@ public void initDefaultCommand() {
      */
     public void spinUp() {
         // Spin up motors
+        RobotMap.rollerSpeedController.set(ROLLER_SPEED);
     }
 
     public void stop() {
-
+        RobotMap.rollerSpeedController.set(0);
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/subsystems/Shooter.java]: There's also a `stopMotor()` method.

**Review by simon-andrews — DISMISSED** [review]
See comments.

GitHub won't let me request changes without a comment.

File: src/org/robockets/steamworks/commands/Shoot.java [hunk 4]
```diff
@@ -11,13 +11,12 @@
 
     public Shoot() {
         addParallel(new SpinSpinners());
-        try {
-            wait(100); // This will be changed
-        } catch (InterruptedException e) {
-            e.printStackTrace();
-        }
+
+        addSequential(new Wait(2)); // This will be changed
```
Comment (bshin100) [inline src/org/robockets/steamworks/commands/Shoot.java]: Wait() is absolutely disgusting. Use WPI's one instead: WaitCommand()

**Comment by UnknownSilicon** [comment]
Tested

File: src/org/robockets/steamworks/PortNumbers.java [hunk 5]
```diff
@@ -10,6 +10,7 @@
     public static final int JOYSTICK_PORT_NUMBER = 0;
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/PortNumbers.java]: PortNumbers is gone.

File: src/org/robockets/steamworks/Robot.java [hunk 6]
```diff
@@ -8,11 +8,11 @@
 import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
-import org.robockets.steamworks.commands.Climb;
-import org.robockets.steamworks.commands.Joyride;
-import org.robockets.steamworks.commands.TunePID;
+import org.robockets.commons.RelativeDirection;
+import org.robockets.steamworks.commands.*;
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/Robot.java]: :expressionless:

File: src/org/robockets/steamworks/Robot.java [hunk 7]
```diff
@@ -8,11 +8,11 @@
...
 import org.robockets.steamworks.subsystems.*;
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/Robot.java]: :expressionless:

File: src/org/robockets/steamworks/Robot.java [hunk 8]
```diff
@@ -47,19 +50,33 @@ public void robotInit() {
...
 		drive = new Joyride(0.5);
     
 		SmartDashboard.putData("Auto mode", chooser);
+
+		SmartDashboard.putData(new SpinSpinners());
+		SmartDashboard.putData(new Shoot());
+
 		SmartDashboard.putData(new Climb(0.5));
 
+		SmartDashboard.putData("MoveConveyorForward", new MoveConveyor(RelativeDirection.YAxis.FORWARD));
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/Robot.java]: Labels could be more clear as to what conveyors/elevators they refer to.

File: src/org/robockets/steamworks/Robot.java [hunk 9]
```diff
@@ -71,6 +88,8 @@ public void robotInit() {
 	@Override
 	public void robotPeriodic() {
 		gearIntake.periodicSmartDashboard();
+		SmartDashboard.putNumber("RawShooterEncoderValue", RobotMap.shooterEncoder.get());
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/Robot.java]: There is now the SDDumper method instead.

File: src/org/robockets/steamworks/commands/ExtraSpace.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+import edu.wpi.first.wpilibj.command.Command;
+import org.robockets.commons.RelativeDirection;
+import org.robockets.steamworks.Robot;
+import org.robockets.steamworks.RobotMap;
+
+/**
+ * @author Jake Backer
+ */
+public class ExtraSpace extends Command {
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/commands/ExtraSpace.java]: "Extra Space" is not a verb.

Comment (simon-andrews) [inline src/org/robockets/steamworks/commands/ExtraSpace.java]: MakeExtraSpace maybe?

**Comment by UnknownSilicon** [comment]
More new code!

**Comment by UnknownSilicon** [comment]
We somehow already have shooter code in drivetrain/master..

**Pull request closed without merging**